### ISSUE TITLE
20220408 broke background color setting

### ISSUE THREAD

**Issue by cranderson** [neomutt/neomutt#3348]
Background colors are always black when they should be white on at least these settings:

```
color body          magenta white "(ftp|http|https|news|nntp|rsync)://[^ ]+" # point out URLs
color body          magenta white [-a-z_0-9.]+@[-a-z_0-9.]+    # e-mail addresses
color header        red white From:
color header        magenta white Subject:
color markers       brightgreen white
color quoted        red white
color tilde         green white
```

Interestingly, these seem to work fine:

```
color header        blue white To:
color hdrdefault    black white
color indicator     white blue
color message       blue white
color normal        blue white
color status        brightyellow blue
```

## Expected Behaviour

The background color should always be white in the screenshot below.

## Actual Behaviour

![Screenshot from 2022-04-10 12-08-55](https://user-images.githubusercontent.com/1354959/162629394-08ef7195-0eea-4a89-8f62-2491f154b9b0.png)

## Steps to Reproduce

All "color" settings:

```
color sidebar_new yellow default
color attachment    magenta white
color body          magenta white "(ftp|http|https|news|nntp|rsync)://[^ ]+" # point out URLs
color body          magenta white [-a-z_0-9.]+@[-a-z_0-9.]+    # e-mail addresses
color bold          brightblue white
color error         red white
color header        red white From:
color header        blue white To:
color header        magenta white Subject:
color hdrdefault    black white
color indicator     white blue
color markers       brightgreen white
color message       blue white
color normal        blue white
color quoted        red white
color search        white red
color signature     yellow white
color status        brightyellow blue
color tilde         green white
color tree          red white
color underline     white blue
```

## How often does this happen?

- Always

## When did it start to happen?

- When I upgraded to neomutt-5:20220408-1.fc34.x86_64 from neomutt-5:20211029-1.fc34.x86_64

## NeoMutt Version

```
NeoMutt 20220408
Copyright (C) 1996-2022 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 5.16.10-100.fc34.x86_64 (x86_64)
ncurses: ncurses 6.2.20200222 (compiled with 6.2.20200222)
libidn2: 2.3.2 (compiled with 2.3.2)
GPGME: 1.15.1
GnuTLS: 3.7.2
libnotmuch: 5.5.0
PCRE2: 10.36 2020-12-04
storage: tokyocabinet, lmdb
compression: lz4, zlib, zstd

Configure options: --sysconfdir=/etc --full-doc SENDMAIL=/usr/sbin/sendmail ISPELL=/usr/bin/hunspell --autocrypt --gnutls --gpgme --gss --disable-idn --idn2 --lmdb --lua --lz4 --notmuch --pcre2 --sasl --tokyocabinet --zlib --zstd

Compilation CFLAGS: -g -O2 -std=c99 -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/usr/include -DNCURSES_WIDECHAR

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  +autocrypt +fcntl -flock -fmemopen +futimens +getaddrinfo +gnutls +gpgme +gss 
  +hcache -homespool +idn +inotify -locales_hack +lua -mixmaster +nls +notmuch 
  -openssl +pcre2 +pgp +sasl +smime +sqlite +sun_attachment 

MAILPATH="/var/mail"
PKGDATADIR="/usr/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

## Extra Info

* Operating System and its version: **Fedora 34**
* Were you using multiple copies of NeoMutt at once?  **No**
* Were you using 'screen' or 'tmux'? **Yes, 'screen', but I reproduced the same issue without 'screen'.  I'm using 'gnome-terminal' then 'ssh' into the remote system running 'neomutt'.  The 'gnome-terminal' and 'ssh' client is Fedora 35.**
* Is your email local (maildir) or remote (IMAP)? **Local maildir**

**Comment by neverpanic** [comment]
I have a similar problem, see
<img width="814" alt="Screen Shot 2022-04-11 at 11 27 33" src="https://user-images.githubusercontent.com/357348/162711104-061cbef9-7827-4ce2-98ab-05fb2947a504.png">

My colorscheme is modified from zenburn:
```
# vim: syntax=muttrc

################
# Object Rules #
################

# Attachments
color attachment color181 color237
# Bold text in mails
color bold color151 color237
# Error messages
color error color228 color95
# Defaut color for headers
color hdrdefault color108 color237
# Various colors for the index...
# ... the message author
color index_author color223 color237 "[^ ]+"
# ... the number of messages in a collapsed thread
color index_collapsed color116 color237
# ... the date of the email
color index_date color248 color237
# ... the flags, uses pattern to match the flags
color index_flags color228 color237 "."
color index_flags color115 color237 "~N|~O"
color index_flags color210 color237 "~F|~D"
# ... the size or linecount of the mail
color index_size color248 color237
# Indicator
color indicator color188 color233
# the "+" markers at the beginning of wrapped lines in the pager (disabled in my config anyway)
color markers color181 color237
# Informational messages
color message color217 color237
# Normal text
color normal color188 color237
# Visual progress bar
color progress color233 color65
# The prompt when typing ':'
color prompt color181 color237
# Quoted text in mails
# Color the first levels of quoted text
color quoted     color108 color237
color quoted1    color116 color237
color quoted2    color247 color237
color quoted3    color108 color237
color quoted4    color116 color237
color quoted5    color247 color237
color quoted6    color108 color237
color quoted7    color116 color237
color quoted8    color247 color237
color quoted9    color108 color237
# Highlight of searched words in the pager
color search color210 color52
# Signature text
color signature color108 color237
# Status lines at the bottom and top
# Underlined text in mail bodies
color underline color151 color237

######################
# Index Object Rules #
######################
# Unread messages in green
color index color115 color237 "~N|~O"
# Deleted messages with red background
color index color210 color237 "~D"
# Flagged messages in red
color index color228 color237 "~F"
# Messages I replied to
color index color108 color237 "~Q"

# =====================================================================
# Header
# =====================================================================
# Colours for items in the reader
color header color115 color236 "^(Date|From|Subject|To):"
color header color187 color236 "^(Bcc|CC|Reply-To|X-Mailer|User-Agent):"

# =====================================================================
# Body
# =====================================================================
# GnuPG output
color body color181 color237 "-----BEGIN PGP SIGNED MESSAGE-----"
color body color181 color237 "-----BEGIN PGP SIGNATURE-----"
color body color181 color237 "-----END PGP SIGNED MESSAGE-----"
color body color181 color237 "-----END PGP SIGNATURE-----"

# Make if very obvious there is a bad signature!
color body color210 color52 "BAD signature .*$"
# Tag a good signature
color body color108 color237 "Good signature .*$"

# Inform the signature can't be verified
color body color217 color237 "Can't check signature: public key not found"
color body color217 color237 "WARNING: .*!"
color body color217 color237 "some signal caught ... exiting"

# highlight diffs
color body color210 color237 "^-.*"
color body color115 color237 "^[+].*"
color body color188 color237 "^diff --git.*"
color body color188 color237 "^index [a-f0-9].*"
color body color228 color237 "^@@.*"

############################################
# Highlights inside the body of a message. #
############################################

# URLs
color body color228 color237 "(http|https|ftp|news|telnet|finger)://[^ \">\t\r\n)]*"
color body color228 color237 "mailto:[-a-z_0-9.]+@[-a-z_0-9.]+"
color body color228 color237 "news:[^ \">\t\r\n]*"
# email addresses
color body color228 color237 "[-a-z_0-9.%$]+@[-a-z_0-9.]+\\.[-a-z][-a-z]+"

###########
# Sidebar #
###########
color sidebar_divider color237 color233
color sidebar_flagged color188 color237
color sidebar_highlight color188 color233
color sidebar_indicator color223 color233
color sidebar_new color115 color237
color sidebar_unread color108 color237
```

I've tried to make sure that everything that could be reasonably used as background here (normal, body, quote) has the correct color (color237) set, but it still ends up being rendered as black.

**Comment by cranderson** [comment]
This issue is definitely similar to #3349.  If I remove all my "mono" settings:

```
mono bold           bold
mono header         bold From:
mono header         bold To:
mono header         bold Subject:
mono indicator      reverse
mono markers        bold
mono quoted         underline
mono search         bold
mono tilde          bold
mono tree           bold
mono underline      underline
```

the behavior changes:

![Screenshot from 2022-04-11 07-33-49](https://user-images.githubusercontent.com/1354959/162731253-57da8854-c6c9-4ea1-9d87-db040ccf4113.png)

It seems the mono settings are being applied when they should only apply if not using a color-capable terminal.
After commenting out all "mono" settings, indicator, markers, quoted, and tilde are all the correct color, and To/Reply-To are correctly no longer bold.  The remaining issue where the background color reverts to black after the end of the printed text in each quoted line still exists.

**Comment by flatcap** [comment]
> mono

ok.  I might need to make `mono` a no-op for now.
Too many people have themes that use it.

> each quoted line still exists.

That should be fixed in the latest push.
I've just rebased [[devel/colour]](https://github.com/neomutt/neomutt/tree/devel/colour) over some other fixes.
The latest commit is 9fcbc1910.
Please can you give it a try.

Thanks

**Comment by LarsHaalck** [comment]
I had issues with foreground colors rather then background colors, but the referenced issue fixes those for me.

**Comment by flatcap** [comment]
Argh!  Bloody `mono` commands!
The one causing your crash is in `/etc/neomuttrc`
That settles it, they have to go.

I've fixed the code 36cd2cb61.

**Comment by cranderson** [comment]
> Argh! Bloody `mono` commands! The one causing your crash is in `/etc/neomuttrc` That settles it, they have to go.
> 
> I've fixed the code [36cd2cb](https://github.com/neomutt/neomutt/commit/36cd2cb61bb6acf40c552b4597bb2cd347df1ce0).

Confirmed that this fixed the segfault with `mono` and the quoted text is also the correct color across the entire terminal line.

It seems /etc/neomuttrc is overriding my own color settings for URLs/email addresses.  How should it behave if the same setting is in /etc/neomuttrc and ~/.muttrc?  I would expect ~/.muttrc to take precedence.

**Comment by flatcap** [comment]
Sorry for the delay, I've been working through all 46 colours on all screens to make sure they blend correctly with the "normal" colour.
I _think_ I've got it!

The normal colour should be merged everywhere and the end-of-line ugliness should be fixed.

Please can you give `master` (e13d420cd) a try.

**Comment by neverpanic** [comment]
End-of-line uglyness is fixed for me in e13d420. Can't comment on the other issues, as I didn't see those.
